src/_pytest/_code/source.py

```diff
@@ -282,14 +284,21 @@ def compile_(  # noqa: F811
     return s.compile(filename, mode, flags, _genframe=_genframe)
 
 
-def getfslineno(obj) -> Tuple[Optional[Union["Literal['']", py.path.local]], int]:
+def getfslineno(obj: Any) -> Tuple[Union[str, py.path.local], int]:
```

Comment:
@bluetech had to take `Any` here instead of `obj`, for the `place_as` attribute check.